ui/raidboss/data/03-hw/dungeon/gubal_library_hard.txt

```diff
@@ -0,0 +1,157 @@
+hideall "--Reset--"
+hideall "--sync--"
+
+0.0 "--Reset--" sync / is no longer sealed/ window 5000 jump 0
+
+# Demon of the Tome
+# -ii 1951 1952
+
+0 "--sync--" sync /The Hall of Magicks will be sealed off/
+6.1 "Triclip" sync /:Demon of the Tome:193A:/
+17.3 "Folio" sync /:Demon of the Tome:193C:/
+23.4 "--sync--" sync /:Top Shelf Tome:193D:/
+24.0 "Book Drop" sync /:Top Shelf Tome:193E:/
+
+29.5 "Triclip" sync /:Demon of the Tome:193A:/
+38.7 "Folio" sync /:Demon of the Tome:193C:/
+44.8 "--sync--" sync /:Top Shelf Tome:193D:/
+45.4 "Book Drop" sync /:Top Shelf Tome:193E:/
+47.0 "Issue" sync /:Middle Shelf Tome:193F:/
+52.6 "Discontinue" sync /:Middle Shelf Tome:1940:/
+58.3 "Frightful Roar" sync /:Demon of the Tome:193B:/
+63.5 "Triclip" sync /:Demon of the Tome:193A:/
+71.5 "Folio" sync /:Demon of the Tome:193C:/
+77.6 "Issue"
+82.2 "Discontinue"
+83.6 "Issue"
+84.2 "Discontinue x3"
+
+102.7 "Triclip" sync /:Demon of the Tome:193A:/ window 15,15
+111.9 "Folio" sync /:Demon of the Tome:193C:/
+118.0 "--sync--" sync /:Top Shelf Tome:193D:/
+118.6 "Book Drop" sync /:Top Shelf Tome:193E:/
+120.2 "Issue" sync /:Middle Shelf Tome:193F:/
+125.8 "Discontinue" sync /:Middle Shelf Tome:1940:/
+131.5 "Frightful Roar" sync /:Demon of the Tome:193B:/
+136.7 "Triclip" sync /:Demon of the Tome:193A:/
+144.7 "Folio" sync /:Demon of the Tome:193C:/
+150.8 "Issue"
+155.4 "Discontinue"
+156.8 "Issue"
+157.4 "Discontinue x3"
+
+# Ordinarily we would sync from the end of the last block
+# However, this tends to break due to multiple identical mechanics close together
+175.9 "Triclip" sync /:Demon of the Tome:193A:/ window 15,15 jump 102.7
+185.1 "Folio"
+191.8 "Book Drop"
+193.4 "Issue"
+199.0 "Discontinue"
+204.7 "Frightful Roar"
+
+# Liquid Flame
+# -ii 1946 1953 194B
+# We use game log lines for the phases because the ACT log lines are unreliable
+# Whichever intermission comes second ends up with phantom "Gains Effect" lines
```

Comment:
I'm not sure I understand the issue here.  Can you give an example of what they look like, or what you mean by phantom?

Reply:
Sure. Let's take Tornado:

```
1200.0 "--sync--" sync /00:....:The liquid flame gains the effect of Anemomorph/ window 200,0
1204.5 "Bibliocide" sync /:Liquid Flame:1945:/ window 10,10
1215.4 "Magnetism/Repel?" sync /:Liquid Flame:194[CD]:/
1226.9 "Bibliocide" sync /:Liquid Flame:1945:/
1237.8 "Magnetism/Repel" sync /:Liquid Flame:194[CD]:/
1246.1 "Form Shift"
```

Unfortunately, there are multiple `gainsEffect` lines in this block:

```
[23:26:48.082] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
[23:26:50.354] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
[23:26:54.011] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
[23:26:56.744] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
[23:27:00.010] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
[23:27:02.919] 1A:4000C430:Liquid Flame gains the effect of Anemomorph from  for 1800.00 Seconds.
```

This seems like it would restrict us to lookbehind windows.  Otherwise we would end up with a constantly shifting timeline during the second phase. However, this phenomenon happens to both Tornado and Hand forms, whichever comes second.

Reply:
Ah, I see.  And every 2 seconds too is extremely rough.

I think the only way you could use the 1A lines here would be to put one sync per potential phase at the end of each phase group, and then jump to the right place.  In other words, N phases have N syncs at the end with small windows but big jumps instead of N phases with 1 sync at the beginning with big windows.

I'm not sure that's worth it to do.  This is probably fine, and since it's an effect it needs to be translated either way.